pootle/apps/pootle_store/views.py

```diff
@@ -362,11 +378,81 @@ def get_units(request):
         'store__translation_project__project',
         'store__translation_project__language',
     )
-    step_queryset = get_step_query(request, units_qs)
```

Comment:
> i need to fix request.profile on export view

the export view uses pootle.core.views.set_permissions which sets request.profile - so need to fix in terms of this PR

however this optimization is a pretty valid one so i have PRed an update to set_permissions here #4500